pkg/pgsql/server/query_machine.go

```diff
@@ -307,30 +308,33 @@ func (s *session) inferParamAndResultCols(statement string) ([]*schema.Column, [
 	if err != nil {
 		return nil, nil, err
 	}
+	// The query string contained in a Parse message cannot include more than one SQL statement;
+	// else a syntax error is reported. This restriction does not exist in the simple-query protocol, but it does exist
+	// in the extended protocol, because allowing prepared statements or portals to contain multiple commands would
+	// complicate the protocol unduly.
 	if len(stmts) > 1 {
-		return nil, nil, ErrMaxStmtNumberExceeded
+		return nil, nil, pserr.ErrMaxStmtNumberExceeded
 	}
 	if len(stmts) == 0 {
-		return nil, nil, ErrNoStatementFound
+		return nil, nil, pserr.ErrNoStatementFound
 	}
 	stmt := stmts[0]
 
-	sel, ok := stmt.(*sql.SelectStmt)
-	if ok != true {
-		return nil, nil, fmt.Errorf("extended query support only query statement")
-
-	}
-	rr, err := s.database.SQLQueryRowReader(sel, true)
-	if err != nil {
-		return nil, nil, err
-	}
-	cols, err := rr.Columns()
-	if err != nil {
-		return nil, nil, err
-	}
 	resCols := make([]*schema.Column, 0)
-	for _, c := range cols {
-		resCols = append(resCols, &schema.Column{Name: c.Selector, Type: c.Type})
+
+	sel, ok := stmt.(*sql.SelectStmt)
+	if ok {
```

Comment:
Just double-checking - this change does enable extended query support for non-query statements, right ?

Reply:
(it's about this commit: https://github.com/codenotary/immudb/pull/840/commits/9cf09489719b6e9d10c207ec73da88b256884d56#diff-3fbf4583347647005ea75130017f5707f4957202e094cdf7849478d81eeeb0d5R326)

Reply:
exactly